command/operator_debug.go

```diff
@@ -1293,22 +1298,155 @@ func (c *OperatorDebugCommand) writeBytes(dir, file string, data []byte) error {
 	return nil
 }
 
-// writeJSON writes JSON responses from the Nomad API calls to the archive
-func (c *OperatorDebugCommand) writeJSON(dir, file string, data interface{}, err error) error {
+// newFilePath returns a validated filepath rooted in the provided directory and
+// path. It has been checked that it falls inside the sandbox and has been added
+// to the manifest tracking.
+func (c *OperatorDebugCommand) newFilePath(dir, file string) (string, error) {
+
+	// Replace invalid characters in filename
+	filename := helper.CleanFilename(file, "_")
+
+	relativePath := filepath.Join(dir, filename)
+	c.manifest = append(c.manifest, relativePath)
+	dirPath := filepath.Join(c.collectDir, dir)
+	filePath := filepath.Join(dirPath, filename)
+
+	// Ensure parent directories exist
+	err := os.MkdirAll(dirPath, os.ModePerm)
```

Comment:
There is also https://github.com/hashicorp/nomad/blob/main/helper/escapingfs/escapes.go#L116

Which also sets permissions to a more reasonable `0755`; `os.ModePerm` is just a mask of permission bits, resulting in `0777`.

Reply:
Fixed in [5423f92](https://github.com/hashicorp/nomad/pull/14610/commits/5423f923b5a30b6f8f0a66a289b32c20b645e3e5), and I've gone thru the rest of the file and fixed the other uses cases of the same thing. 

(I also fixed all the places we were doing `\"%s\"` in error messages while I was in there 😁 )